### ISSUE TITLE
[Bug] ie_api.pyx:357: RuntimeError: dimension (4) in node dim must be a non-negative integer: at offset

### ISSUE THREAD

**Issue by vixadd** [openvinotoolkit/openvino#11351]
##### System information (version)
- OpenVINO => 2021.4
- Operating System / Platform => Ubuntu 20.04
- Compiler => gcc/g++ (GNU)
- Problem classification: IECore (read_network) Model Instantiation
- Framework: ONNX (if applicable)
- Model name: Custom Model

##### Detailed description
I have a custom-built ONNX model that I've created using tools that come from OpenVino 2021.4 version. This custom model has worked for the past 3 months without any problems. In the past 2 weeks, we've started to see an error where sometimes the model.xml and the model.bin files will be sufficient to call the IECore.read_network function on and sometimes it isn't. So the two possible outcomes are:
 * The model works fine and we see no issues.
 * The model fails, and we see the error mentioned below.

Keep in mind, nothing changes with respect to the XML or BIN file. The envrionment doesn't change. The software path for the inference engine doesn't change all that much. We're not instantiating IECore any differently. - Other than that we first saw this error in pytest.
```bash
program.py:46: in __init__
    self.model = self.ie.read_network(model_xml, model_bin)
ie_api.pyx:357: in openvino.inference_engine.ie_api.IECore.read_network
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   RuntimeError: dimension (4) in node dim must be a non-negative integer: at offset

ie_api.pyx:400: RuntimeError
--------------------------------------------------------------------------------------------- Captured stderr call ---------------------------------------------------------------------------------------------
[ INFO ]  Not converting, graph optimizing, or quantizing: using existing IR files at %s and %s
[ INFO ]  Loading model from IR files at %s and %s
2022-03-30 16:21:49.069372: I tensorflow/core/platform/cpu_feature_guard.cc:151] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2022-03-30 16:21:49.074407: I tensorflow/core/common_runtime/process_util.cc:146] Creating new thread pool with default inter op setting: 
---------------------------------------------------------------------------------------------- Captured log call -----------------------------------------------------------------------------------------------
INFO     program.py:225 Not converting, graph optimizing, or quantizing: using existing IR files at /tmp/model.xml and /tmp/model.bin
INFO     program.py:229 Loading model from IR files at /tmp/model.xml and /tmp/model.bin
```
I mainly want to make sure that I'm not missing something as far as usage of the framework goes. This framework has been performing perfectly for months with no hitches, so I'm a little confused as to why I'm only seeing this now.

##### Steps to reproduce
You may see this error pop up only in pytest. Or with the most recent changes to openvino. We installed the framework from APT, and we're using the most up-to-date patch.

##### Issue submission checklist

 - [ x ] I report the issue, it's not a question
   
 - [ x ] I checked the problem with documentation, FAQ, open issues, Stack Overflow, etc and have not found solution

**Comment by vixadd** [comment]
**UPDATE**
 - [ x ] There is reproducer code and related data files: images, videos, models, etc.

I'm seeing the error when I try to work with this onnx file and do the model optimization on it.

```python
class Module:
    def __init__(self, ie, model):
        self.ie = ie
        self.model = ie.read_network(model, model.replace('.xml', '.bin'))
        self.active_requests = 0
        self.clear()

    def deploy(self, device, plugin_config, max_requests=1):
        self.max_requests = max_requests
        self.exec_net = self.ie.load_network(self.model, device, config=plugin_config, num_requests=max_requests)

    def enqueue(self, inp):
        self.clear()

        if self.max_requests <= self.active_requests:
            logging.warning('Processing request rejected - too many requests')
            return False

        self.exec_net.start_async(self.active_requests, inp)
        self.active_requests += 1
        return True

    def wait(self):
        if self.active_requests <= 0:
            return

        self.perf_stats = [None, ] * self.active_requests
        self.outputs = [None, ] * self.active_requests
        for i in range(self.active_requests):
            self.exec_net.requests[i].wait()
            self.outputs[i] = self.exec_net.requests[i].output_blobs
            self.perf_stats[i] = self.exec_net.requests[i].get_perf_counts()

        self.active_requests = 0

    def get_outputs(self):
        self.wait()
        return self.outputs

    def get_performance_stats(self):
        return self.perf_stats

    def clear(self):
        self.perf_stats = []
        self.outputs = []

    def infer(self, inputs):
        self.clear()
        self.start_async(*inputs)
        return self.postprocess()
```

```python
from openvino.inference_engine import IECore

class ResNetDetect(Module):
    
    class Result:
        def __init__(self, output):
            self.output = output

    
    def __init__(self, ie, model, input_shape):
        super(ResNetDetect, self).__init__(ie, model)
        
        assert len(self.model.input_info) == 1, "Expected only 1 input blob."

        self.input_blob = next(iter(self.model.input_info))
        self.input_shape = self.model.input_info[self.input_blob].input_data.shape
        
    def preprocess(self, xin):
        return xin # TODO
    
    def start_async(self, xin):
        inp = self.preprocess(xin)
        self.enqueue(inp)
    
    def enqueue(self, xin):
        return super(ResNetDetect, self).enqueue({self.input_blob: xin})
    
    def postprocess(self):
        outputs = self.get_outputs()
        results = []

        for output in outputs:
            result = ResNetDetect.Result(output)
            
            # TODO: Post-process results ???
            
            results.append(result)
        
        return results

        
class ResNet50Infer:
    CPU_DEVICE = 'CPU'
    MAX_REQUESTS = 100
    
    def __init__(self, openvino_xml, openvino_bin, perf_count=False, num_threads=0):
        self.xml_file   = openvino_xml
        self.bin_file   = openvino_bin
        self.ie         = IECore()
        self.perf_count = perf_count
        self.num_threads  = num_threads
        
        self.net_detect = ResNetDetect(self.ie, self.xml_file, [1,3,224,224])
        
        # Deploy the model engine.
        device = ResNet50Infer.CPU_DEVICE
        self.net_detect.deploy(device, self.get_config(device), ResNet50Infer.MAX_REQUESTS)
        
    def get_config(self, device):
        """
        https://docs.openvinotoolkit.org/latest/openvino_docs_IE_DG_supported_plugins_CPU.html#supported_configuration_parameters
        """
        config = {
            "PERF_COUNT": "YES" if self.perf_count else "NO",
            "CPU_THREADS_NUM": str(self.num_threads), # 0 - Default Using all logical cores.
#             "CPU_BIND_THREAD": "YES",
#             "CPU_THROUGHPUT_STREAMS": str(1),
        }
        
        if device == "GPU" and self.gpu_lib:
            config["CONFIG_FILE"] = self.gpu_lib
        
        return config

    def inference(self, xin):
        return self.net_detect.infer((xin,))
```
I can't attach the onnx file for some reason. I think you can download a ResNet50 onnx file.

**Comment by vixadd** [comment]
**ANOTHER UPDATE - POSSIBLE SOLUTION**

If I remove my tensorflow import at the top of the file, everything works again. It seems that the conda intel channel tensorflow has been updated in the past few days. I think this is still a bug because I need tensorflow in the framework, but there seems to be an issue with importing tensorflow and running the read_network function.

Openvino Devs: Check to see if there's some sort of memory leak that's caused with importing tensorflow.
The way I install Openvino:
```
RUN curl https://apt.repos.intel.com/openvino/2021/GPG-PUB-KEY-INTEL-OPENVINO-2021 -O
RUN sudo -E apt-key add GPG-PUB-KEY-INTEL-OPENVINO-2021
RUN sudo -E echo "deb https://apt.repos.intel.com/openvino/2021 all main" | \
    sudo -E tee /etc/apt/sources.list.d/intel-openvino-2021.list
RUN sudo -E apt update -y
RUN sudo -E apt install -y $(apt-cache search intel-openvino-dev-ubuntu20 | grep ${INTEL_VER} | awk 'END {print $1}')

RUN sudo -E chown -R vix:vix /opt/intel/

RUN --mount=type=secret,mode=<> \
    --mount=type=secret,mode=<> \
    cd /opt/intel/openvino_2021/deployment_tools/tools/post_training_optimization_toolkit && \
    /opt/conda/envs/intel/bin/pip3 install -r ./requirements.txt && \
    cd -;

RUN --mount=type=secret,mode=<> \
    --mount=type=secret,mode=<> \
    cd /opt/intel/openvino_2021/deployment_tools/model_optimizer && \
    /opt/conda/envs/intel/bin/pip3 install -r ./requirements_onnx.txt && \
    cd -;

```
The way I install tensorflow:
```
RUN conda install -n intel -c intel -y tensorflow mkl-include ipp-include intel-fortran-rt tbb-devel
```

**Comment by ilya-lavrenov** [comment]
@slyubimt could you please have a look?

**Comment by vixadd** [comment]
Thank you for taking a look @slyubimt - I'm using Tensorflow v. 2.8.0 from the intel channel on conda.

**Comment by slyubimt** [comment]
@andrei-kochin , @akuporos  Can you look at this issue from the perspective of used tensorflow version?

@vixadd Can you provide more info on Python version used and IR version of the model?

**Comment by vixadd** [comment]
Python version is 3.8.11. Using intel-python 
[GCC 9.3.0] :: Intel Corporation on linux.

IR Version of OpenVino: At the top of my XML file it has:
```xml
<net name="model" version="10">
```
I assume that means IR version 10.

**Comment by jgespino** [comment]
@vixadd 

> <net name="model" version="10">
> I assume that means IR version 10.

Yes, the models generated by OpenVINO 2021.4.2 are IR version 10.

> Downgrading to 2.7.0 solves this issue.

Glad you were able to get the issue resolved with TF 2.7.0. 

I will close this issue, OpenVINO 2021.4.2 installs TF 2.4.4 and the newer OpenVINO 2022.1 installs TF 2.5.3. If you continue to run into issues, I recommend the Tensorflow version installed with OpenVINO
